crates/rooch/src/tx_runner.rs

```diff
@@ -199,3 +264,90 @@ fn convert_to_verified_tx(
         verified_action,
     ))
 }
+
+pub fn dry_run_tx_locally(
+    client: Client,
+    tx: RoochTransactionData,
+) -> anyhow::Result<Option<DryRunTransactionResponseView>> {
+    let state_root =
+        get_latest_state_root(&client)?.expect("getting the latest state root is failed");
+    let (_, raw_transaction_output, error_info_opt) = execute_tx_locally(state_root, client, tx)?;
+
+    match error_info_opt {
+        Some(vm_error_info) => {
+            let raw_output_view = RawTransactionOutputView {
+                status: KeptVMStatusView::from(raw_transaction_output.status),
+                gas_used: StrView::from(raw_transaction_output.gas_used),
+                is_upgrade: false,
+            };
+            Ok(Some(DryRunTransactionResponseView {
+                raw_output: raw_output_view,
+                vm_error_info,
+            }))
+        }
+        None => Ok(None),
+    }
+}
+
+fn get_latest_state_root(client: &Client) -> anyhow::Result<Option<Vec<u8>>> {
+    tokio::task::block_in_place(|| {
+        Handle::current().block_on(async {
+            let sequencer_order = client.rooch.sequencer_order().await?;
+            let transaction = client
+                .rooch
+                .get_transactions_by_order(Some(sequencer_order), Some(1), Some(false))
+                .await?;
```

Comment:
The `rooch_status` RPC function returns the latest state root. We can use the function; we do not need the `sequencer_order` function.

Reply:
Fixed